mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -87,6 +87,14 @@ final class DecisionTreeRegressor @Since("1.4.0") (@Since("1.4.0") override val
     trees.head.asInstanceOf[DecisionTreeRegressionModel]
   }
 
+  /** (private[ml]) Train a decision tree on an RDD */
+  private[ml] def train(data: RDD[LabeledPoint],
```

Comment:
GBTs in spark.ml are handled by converting a dataframe to an RDD of `LabeledPoint` and then working with that during training. I added a new `train` method to accept an RDD that can be used to train the trees in the GBT ensemble. I appreciate feedback on this approach or alternative approaches.